src/main/java/frc/robot/subsystems/drivetrain/Drivetrain.java

```diff
@@ -0,0 +1,172 @@
+package frc.robot.subsystems.drivetrain;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.kinematics.DifferentialDriveKinematics;
+import edu.wpi.first.wpilibj.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.wpilibj.kinematics.DifferentialDriveWheelSpeeds;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.components.Pigeon;
+import frc.robot.subsystems.MoveableSubsystem;
+import static frc.robot.Robot.robotConstants;
+
+public class Drivetrain extends SubsystemBase implements MoveableSubsystem {
+
+  private DifferentialDriveKinematics kinematics;
+  private DifferentialDriveOdometry odometry;
+
+  private WPI_TalonFX leftRearTalon;
+  private WPI_TalonFX leftMiddleTalon;
+  private WPI_TalonFX leftFrontTalon;
+  private WPI_TalonFX rightRearTalon;
+  private WPI_TalonFX rightMiddleTalon;
+  private WPI_TalonFX rightFrontTalon;
+  private WPI_TalonSRX rightEncoder;
+  private WPI_TalonSRX leftEncoder;
+  private Pigeon gyro;
+
+  private DifferentialDrive drivetrain;
+/**
+ * This is the subsystem of the drivetrain
+ */
+  public Drivetrain() {
+    leftRearTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_LEFT_REAR_TALON);
+    leftMiddleTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_LEFT_MIDDLE_TALON);
+    leftFrontTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_LEFT_FRONT_TALON);
+    rightRearTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_RIGHT_REAR_TALON);
+    rightMiddleTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_RIGHT_MIDDLE_TALON);
+    rightFrontTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_RIGHT_FRONT_TALON);
+    leftMiddleTalon.follow(leftFrontTalon);
+    leftRearTalon.follow(leftFrontTalon);
+    rightMiddleTalon.follow(rightFrontTalon);
+    rightRearTalon.follow(rightFrontTalon);
+    drivetrain = new DifferentialDrive(leftFrontTalon, rightFrontTalon);
+    drivetrain.setDeadband(0);
+    // TODO: set correct port for pigeon gyro.
+    gyro = new Pigeon(robotConstants.can.DRIVETRAIN_LEFT_REAR_TALON);
+    kinematics = new DifferentialDriveKinematics(robotConstants.drivetrainConstants.WHEEL_BASE_WIDTH);
+    odometry = new DifferentialDriveOdometry(Rotation2d.fromDegrees(getAngle()));
+    // TODO: set correct talons for encoders.
+    leftEncoder = new WPI_TalonSRX(robotConstants.can.TEMPORARY_TALON_FOR_LEFT_DRIVETRAIN_ENCODER);
+    rightEncoder = new WPI_TalonSRX(robotConstants.can.TEMPORARY_TALON_FOR_RIGHT_DRIVETRAIN_ENCODER);
+  }
+
```

Comment:
etc.
    gyro = new Pigeon(robotConstants.can.DRIVETRAIN_LEFT_REAR_TALON);
    kinematics = new DifferentialDriveKinematics(robotConstants.drivetrainConstants.WHEEL_BASE_WIDTH);
    odometry = new DifferentialDriveOdometry(Rotation2d.fromDegrees(getAngle()));
    // TODO: set correct talons for encoders.
    leftEncoder = new WPI_TalonSRX(robotConstants.can.TEMPORARY_TALON_FOR_LEFT_DRIVETRAIN_ENCODER);
    rightEncoder = new WPI_TalonSRX(robotConstants.can.TEMPORARY_TALON_FOR_RIGHT_DRIVETRAIN_ENCODER);

order it like the order up (before constractor)